docs/03-deploy.Rmd

```diff
@@ -178,3 +178,74 @@ deploy:
 This means the website will be pushed to the `master` branch of the repository `username/username.github.io` (remember to replace `username` with your actual username).
 
 You can also deploy your website to Amazon S3\index{Amazon S3}, and the setup on the R side is very similar to what we have introduced for GitHub Pages. The only difference is in the last step, where you change the target from GitHub Pages to Amazon S3. For more information, please see the documentation on Travis: https://docs.travis-ci.com/user/deployment/s3/.
+
+## GitLab Pages
+
+[GitLab](http://gitlab.com/) is a very popular way to host the source code of your project.
+GitLab has a [built in Continuous Integration & Deployment service](https://about.gitlab.com/features/gitlab-ci-cd/)
+that can be used to host static websites, named [GitLab Pages](https://about.gitlab.com/features/pages/).
+The major advantage of using GitLab Pages is that
+GitLab Continuous Integration & Deployment service will compile all your Rmd posts
+instead of your local computer
+**and any generated content, such as HTML files, will be stored in the Git repository**.
+
+Before I show you how, I’d like to mention two issues that you should be aware of:
+
+- It is somewhat unpredictable the output
+  because it is fully automatic
+  and you do not have a chance to see the new content or results to be published.
+  There are many factors that could affect building the site:
+  the R version,
+  availability of certain R packages,
+  system dependencies, and network connection, etc.
+- The time required to compile all Rmd files may be very long
+  and cause timeouts on GitLab,
+  depending on how time-consuming your R code is.
+
+The first problem can be solved by specifying the version of the package you are using
+on the installation instructions.
+The second problem can be solved by using caching mechanism.
+
+Like GitHub Pages,
+you can have project websites in separate repositories,
+e.g., you may have two independent websites `https://username.gitlab.io/proj-a/` and `https://username.gitlab.io/proj-b/`,
+corresponding to GitLab repositories `https://gitlab.com/username/proj-a/` and `https://gitlab.com/username/proj-b/`, respectively.
+Is also possible to associate each repository with a domain or subdomain name,
+for example,
+you may replace use `https://proj-a.username.com/` and `https://proj-b.username.com/`.
+
+GitLab Continuous Integration & Deployment service uses the instructions stored on the file `.gitlab-ci.yml` at the root of the repository.
+Here is the configuration file, `.gitlab-ci.yml`, for https://rgaiacs.gitlab.io/blogdown-example:
+
+```
+image: debian:buster-slim
+
+before_script:
+  - apt-get update
+  - apt-get -y install curl libcurl4-openssl-dev openssl libssl-dev git pandoc r-base-core r-recommended
```

Comment:
If you install blogdown from CRAN, I guess you won't need `curl libcurl4-openssl-dev openssl libssl-dev git`. The `r-recommended` package is not necessary, either.

Reply:
When I was testing last month I got a error mentioning that `curl` was missing. So I suggest to keep it. I removed `r-recommended`. Fixed at cc3c8e1.

Reply:
Interesting. I need to try this by myself. My next comment said that you could install blogdown from CRAN, so you wouldn't need to install devtools. rmarkdown and stringi are dependencies of blogdown, so they don't need to be installed explicitly, either.

Reply:
I updated the PR to remove `devtools`, `rmarkdown` and `stringi`. I don't remember why I added `rmarkdown` but I remember that sometimes `stringi` wasn't installed for some unknown reason and the recommendation based on StackOverflow was to install it explicitly.

Reply:
Thanks! Could you actually try this in your repo and make sure it works? stringi should be automatically installed since stringr depends on it, and knitr depends on stringr, and rmarkdown depends on knitr, and blogdown depends on rmarkdown.

Reply:
Or add me as a collaborator to your GitLab repo, so I can test it by myself. My GitLab username is `xie`.

Reply:
> Could you actually try this in your repo and make sure it works?

Waiting GitLab build.

> Add me as a collaborator to your GitLab repo.

Done.

Reply:
Thanks! It seems to work with only three packages pandoc, r-base, and blogdown.

Reply:
FYI I renamed the repo to `blogdown-gitlab`, which is more specific, since `blogdown-example` sounds too vague. You can update the git remote by:

```bash
git remote set-url origin git@gitlab.com:rgaiacs/blogdown-gitlab.git
```